Draft-Accepted/RFCNNNN-PathUpdateForPackageInstall.md

```diff
@@ -0,0 +1,189 @@
+     - WT response:
+       > clearing the list will change it for cmd as well, of course. A windows upgrade might restore part of the list (the three entries we know about)
+       > - we migrate `InteractiveExtensionLevel` and the package manager info from build to build
+       > - however, I don't know if migration manifests account for deletions of keys. that's where it gets weird for me, and that's where I don't actually know
+     - [dongbow && jason]'s thoughts on this comment
+       - It's unlikely that a user wants to use different settings for that list between PowerShell and CMD.
+       - Windows upgrade shouldn't mess up with the changes a user made to those registry keys. It sounds like a bug if it does.
+       - Even if deleted key will be restored with the default value by windows upgrade, clearing the list but keeping the key around should be fine with windows upgrade.
+
+4. What about `Start-Process`, should we do the detection within that cmdlet too?
+
+   - **[PS REVIEW COMMENT]** No, the feature is only for the native command processor.
+
+   - **_[WT comments]_** CMD only covers EXE, but `scoop` is a `.ps1` or `.bat` file, so CMD doesn't see it. PowerShell may want to broaden your support to account for any command.
+     - [dongbow && jason]'s thoughts on this comment
+       - [Scoop](https://scoop.sh/) creates shims inside the `~\scoop\shims` folder for terminal applications, which is accessible in the PATH. So `Scoop` doesn't need PATH refreshing.
+       - We should scope the work to native command only. Supporting non-native commands would need broader changes outside of the native command processor without a clear gain.
+
+5. The fix to CMD checks command name only, so it will try refreshing `PATH` even if `winget` or `choco` are invoked for purpose other than installing a new package, e.g. `winget search`.
+Should we optimize it in PowerShell to only refresh `PATH` when `winget install` or `choco install` actually runs?
+
+   - **[PS REVIEW COMMENT]** No, make it simple.
+
+6. Refreshing PATH could cause a problem in command resolution -- a command that could be found in an unloaded module may be resolved to a native command due to the appending of a new path to `PATH`. How much should we care about this situation? (it's an existing issue anyway)
+
+   - [dongbow] IMHO, this should not be a concern for script execution/automation, as it should use module-name-qualified command name.
+   I also don't think it's a real problem to interactive usage -- if the user run `foo` after `winget install foo`, they would expect the installed tool to be invoked instead of a function named `foo` from an unloaded module.
+
+   - **[PS REVIEW COMMENT]** Not concerned about this except for the WDAC and AppLocker scenario (see below).
+
+### WDAC and AppLocker scenario
+
+- If `Env` provider is disabled or read-only, the feature should be disabled.
```

Comment:
It looks weird. And how cmd.exe disable this feature? And is it even worth adding this switch? If so, why isn't it a key in the registry or an option in the config?

Reply:
cmd.exe disables the feature when
1. The list of package managers is cleared (which are defined as subkeys under `HKLM\SOFTWARE\Microsoft\Command Processor\KnownPackageManagers`)
2. The `HKLM\SOFTWARE\Microsoft\Command Processor\InteractiveExtensionLevel` is set to `0`. This is a special key for cmd.exe -- it disables all new interactive features introduced to cmd.exe, the "Path Update" is one of them.